web/ui/src/Services/ServiceDetailsController.js

```diff
@@ -3,1076 +3,1107 @@
-                if(!name || !name.length){
-                    return "Missing Name";
+        ["$scope", "$q", "$routeParams", "$location", "resourcesFactory",
+            "authService", "$modalService", "$translate", "$notification",
+            "$timeout", "miscUtils", "hostsFactory", "$serviceHealth", "Service",
+            "poolsFactory", "CCUIState", "$cookies", "areUIReady", "LogSearch",
+            function ($scope, _$q, $routeParams, $location, _resourcesFactory,
+                authService, $modalService, $translate, _$notification,
+                $timeout, _utils, hostsFactory, _serviceHealth, Service,
+                poolsFactory, CCUIState, $cookies, areUIReady, LogSearch) {
+
+                // api access via angular context
+                $notification = _$notification;
+                serviceHealth = _serviceHealth;
+                $q = _$q;
+                resourcesFactory = _resourcesFactory;
+                utils = _utils;
+
+                // Ensure logged in
+                authService.checkLogin($scope);
+                $scope.hostsFactory = hostsFactory;
+
+                $scope.defaultHostAlias = $location.host();
+                if (utils.needsHostAlias($location.host())) {
+                    resourcesFactory.getHostAlias().success(function (data) {
+                        $scope.defaultHostAlias = data.hostalias;
+                    });
                 }
 
-                // if name already exists
-                for (var i in $scope.publicEndpoints.data) {
-                    if (name === $scope.publicEndpoints.data[i].Name) {
-                        return "Name already exists: "+ newPublicEndpoint.name;
-                    }
-                }
+                //add Public Endpoint data
+                $scope.publicEndpoints = { add: {} };
+
+                //add service endpoint data
+                $scope.exportedServiceEndpoints = {};
+
+                $scope.click_pool = function (id) {
+                    resourcesFactory.routeToPool(id);
+                };
+
+                $scope.click_host = function (id) {
+                    resourcesFactory.routeToHost(id);
+                };
+
+                $scope.modalAddPublicEndpoint = function () {
+
+                    areUIReady.lock();
+                    $scope.protocols = [];
+                    $scope.protocols.push({ Label: "HTTPS", UseTLS: true, Protocol: "https" });
+                    $scope.protocols.push({ Label: "HTTP", UseTLS: false, Protocol: "http" });
+                    $scope.protocols.push({ Label: "Other, secure (TLS)", UseTLS: true, Protocol: "" });
+                    $scope.protocols.push({ Label: "Other, non-secure", UseTLS: false, Protocol: "" });
+
+                    // default public endpoint options
+                    $scope.publicEndpoints.add = {
+                        type: "port",
+                        app_ep: $scope.currentService.exportedServiceEndpoints[0],
```

Comment:
similar situation here. the endpoints object created here was existing code.  Since there are many tie-ins to templates, etc. I did not change the object property name.